src/nix/store-check-trusted.cc

```diff
@@ -0,0 +1,61 @@
+#include <sstream>
+
+#include "command.hh"
+#include "logging.hh"
+#include "serve-protocol.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include "local-fs-store.hh"
+#include "util.hh"
+#include "worker-protocol.hh"
+
+using namespace nix;
+
+namespace {
+
+bool checkPass(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_GREEN "[PASS] " ANSI_NORMAL + msg);
+    return true;
+}
+
+bool checkFail(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_RED "[FAIL] " ANSI_NORMAL + msg);
+    return false;
+}
+
+}
+
+struct CmdCheckTrusted : StoreCommand
+{
+    bool success = true;
+
+    Category category() override { return catUtility; }
+
+    std::string description() override
+    {
+        return "ask a Nix store if you are a trusted-user, return exit code 0 if trusted, 1 if not trusted, 2 if unexpected error";
+    }
+
+    void run(ref<Store> store) override
+    {
+        logger->log("Running checks against store uri: " + store->getUri());
+        success &= checkTrustedUser(store);
```

Comment:
```suggestion
        bool success = checkTrustedUser(store);
```

Unless there's a good reason to make it a class variable, I'd rather keep it local as it makes things much more readable

Reply:
This is a pattern I copied from doctor.cc, nothing intentional.